internal/auth/telemetry_claim.go

```diff
@@ -51,3 +53,36 @@ func getTelemetryClaim(ctx context.Context) (*TelemetryClaim, error) {
 	}
 	return claim, nil
 }
+
+func ValidAttestationClaim(ctx context.Context, filter *model.AttestationFilter) bool {
```

Comment:
Maybe the name could hint at the "hey is this filter okay" nature of this as well. I get that the filter is an argument so maybe... it just feels off.